### PR TITLE
Additions and cleanups for contrib/debian/copyright

### PR DESCRIPTION
**Pull request by daira**
This solves some syntax issues and adds missing copyright entries, including entries for Rust dependencies that cannot be Expat/MIT-licensed.

### CODE REVIEW

**Review by nuttycom — APPROVED** [review]
ACK

File: contrib/debian/copyright [hunk 1]
```diff
@@ -71,12 +71,33 @@ Files: build-aux/m4/ax_check_compile_flag.m4
  build-aux/m4/ax_check_preproc_flag.m4
 Copyright: 2008, Guido U. Draheim <guidod@gmx.de>
  2011, Maarten Bosmans <mkbosmans@gmail.com>
-License: GPL-3+ with Autoconf exception
+License: GPL-3+-with-Autoconf-exception
```
Comment (str4d) [inline contrib/debian/copyright:74]: NACK. This undoes the changes I made in https://github.com/zcash/zcash/pull/5333/commits/a672318b4d9341a2ac3938f2327a3f218ad88ae8, which were specifically to ensure we followed the v1 format of the Debian copyright file. Note that it says "License names are case-insensitive, and may not contain spaces"; here the license name is `GPL-3+`, and the Autoconf exception is applied to the license with spaces (exactly matching the examples given in the v1 specification).

**Review by str4d — CHANGES_REQUESTED** [review]

File: contrib/debian/copyright [hunk 2]
```diff
@@ -1158,9 +1261,6 @@ License: GPL-3+ with Autoconf exception
  You should have received a copy of the GNU General Public License along
  with this program. If not, see <http://www.gnu.org/licenses/>.
  .
- On Debian systems, the full text of the GNU General Public License
- version 3 can be found in the file `/usr/share/common-licenses/GPL-3'.
- .
```
Comment (str4d) [inline contrib/debian/copyright:1163]: NACK. I added this in https://github.com/zcash/zcash/pull/5333/commits/a672318b4d9341a2ac3938f2327a3f218ad88ae8, again to follow the Debian copyright file v1 format.

**Review by str4d — CHANGES_REQUESTED** [review]

File: contrib/debian/copyright (hunk 1 above)
Comment (daira) [reply contrib/debian/copyright]: Removed that commit.

File: contrib/debian/copyright (hunk 2 above)
Comment (daira) [reply contrib/debian/copyright]: Removed this commit.

File: src/rust/include/tracing.h [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// Copyright (c) 2020 Jack Grigg
+// Copyright (c) 2020-2021 The Zcash developers
```
Comment (str4d) [inline src/rust/include/tracing.h:1]: 👍

**Review by str4d — APPROVED** [review]
ACK. I did not check the source licenses, but the changes to the Debian copyright file LGTM.

**Pull request merged**